File: src/main/java/dev/lbuddyboy/flash/user/model/Demotion.java
package dev.lbuddyboy.flash.user.model;

import dev.lbuddyboy.flash.FlashLanguage;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.text.SimpleDateFormat;
import java.util.TimeZone;

@AllArgsConstructor
@Getter
public class Demotion {

    private final String demotedFrom;
    private final long demotedAt;

    public String getDemotedAtDate() {
        SimpleDateFormat sdf = new SimpleDateFormat();
        sdf.setTimeZone(TimeZone.getTimeZone(FlashLanguage.TIMEZONE.getString()));
        return sdf.format(demotedAt);
    }

}

File: src/main/java/dev/lbuddyboy/flash/user/model/Demotion.kt
package dev.lbuddyboy.flash.user.model

import dev.lbuddyboy.flash.FlashLanguage
import lombok.AllArgsConstructor
import lombok.Getter
import java.text.SimpleDateFormat
import java.util.*

@AllArgsConstructor
@Getter
class Demotion {
    private val demotedFrom: String? = null
    private val demotedAt: Long = 0
    fun getDemotedAtDate(): String {
        val sdf = SimpleDateFormat()
        sdf.timeZone = TimeZone.getTimeZone(FlashLanguage.TIMEZONE.string)
        return sdf.format(demotedAt)
    }
}